lib/galaxy/managers/genomes.py

```diff
@@ -64,3 +71,30 @@ def _get_index_filename(self, id, tbl_entries, ext, index_type):
             raise ReferenceDataError(f"Data tables not found for {index_type} for {id}")
         else:
             return f"{file_name}{ext}"
+
+
+class GenomeFilterMixin:
+    orm_filter_parsers: "OrmFilterParsersType"
+    valid_ops = ("eq", "contains", "has")
+
+    def create_genome_filter(self, attr, op, val):
+        def _create_genome_filter(model_class=None):
+            if op not in GenomeFilterMixin.valid_ops:
+                raise_filter_err(attr, op, val, "bad op in filter")
+            if model_class is None:
+                return True
+            # Doesn't filter genome_build for collections
+            if model_class.__name__ == "HistoryDatasetCollectionAssociation":
+                return False
+            column = func.json_extract(model_class.table.c._metadata, "$.dbkey")
```

Comment:
I think json_extract is specific to sqlite ? 

```
UndefinedFunction: function json_extract(bytea, unknown) does not exist
LINE 4: ...taset_association.history_id = 7402498 AND (lower(json_extra...
                                                             ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

  File "sqlalchemy/engine/base.py", line 1905, in _execute_context
    self.dialect.do_execute(
  File "sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
ProgrammingError: (psycopg2.errors.UndefinedFunction) function json_extract(bytea, unknown) does not exist
LINE 4: ...taset_association.history_id = 7402498 AND (lower(json_extra...
                                                             ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

[SQL: SELECT anon_1.history_id AS anon_1_history_id, anon_1.history_content_type AS anon_1_history_content_type, anon_1.id AS anon_1_id, anon_1.type_id AS anon_1_type_id, anon_1.hid AS anon_1_hid, anon_1.extension AS anon_1_extension, anon_1.dataset_id AS anon_1_dataset_id, anon_1.collection_id AS anon_1_collection_id, anon_1.name AS anon_1_name, anon_1.state AS anon_1_state, anon_1.size AS anon_1_size, anon_1.deleted AS anon_1_deleted, anon_1.purged AS anon_1_purged, anon_1.visible AS anon_1_visible, anon_1.create_time AS anon_1_create_time, anon_1.update_time AS anon_1_update_time 
FROM (SELECT history_dataset_association.history_id AS history_id, %(param_1)s AS history_content...
  File "starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "starlette_context/middleware/raw_middleware.py", line 93, in __call__
    await self.app(scope, receive, send_wrapper)
  File "starlette/middleware/base.py", line 108, in __call__
    response = await self.dispatch_func(request, call_next)
  File "galaxy/webapps/galaxy/fast_app.py", line 103, in add_x_frame_options
    response = await call_next(request)
  File "starlette/middleware/base.py", line 84, in call_next
    raise app_exc
  File "starlette/middleware/base.py", line 70, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "starlette/routing.py", line 66, in app
    response = await func(request)
  File "fastapi/routing.py", line 241, in app
    raw_response = await run_endpoint_function(
  File "fastapi/routing.py", line 169, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "galaxy/webapps/galaxy/api/history_contents.py", line 469, in index
    items = self.service.index(
  File "galaxy/webapps/galaxy/services/history_contents.py", line 315, in index
    return self.__index_v2(trans, history_id, params, serialization_params, filter_query_params, accept)
  File "galaxy/webapps/galaxy/services/history_contents.py", line 979, in __index_v2
    contents = self.history_contents_manager.contents(
  File "galaxy/managers/history_contents.py", line 125, in contents
    return self._union_of_contents(
  File "galaxy/managers/history_contents.py", line 259, in _union_of_contents
    contents_results = self._union_of_contents_query(container, **kwargs).all()
  File "sqlalchemy/orm/query.py", line 2773, in all
    return self._iter().all()
  File "sqlalchemy/orm/query.py", line 2916, in _iter
    result = self.session.execute(
  File "sqlalchemy/orm/session.py", line 1717, in execute
    result = conn._execute_20(statement, params or {}, execution_options)
  File "sqlalchemy/engine/base.py", line 1710, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "sqlalchemy/engine/base.py", line 1577, in _execute_clauseelement
    ret = self._execute_context(
  File "sqlalchemy/engine/base.py", line 1948, in _execute_context
    self._handle_dbapi_exception(
  File "sqlalchemy/engine/base.py", line 2129, in _handle_dbapi_exception
    util.raise_(
  File "sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "sqlalchemy/engine/base.py", line 1905, in _execute_context
    self.dialect.do_execute(
  File "sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)

```

from https://sentry.galaxyproject.org/share/issue/c069ae6b4a084277b4fabbc93a928d6c/

Reply:
Ah i see, trying a workaround for postgresql here

Reply:
fwiw, json_extract is not specific to sqlite as it exists in mysql (which is probs why some admins don't have the error). But it doesn't exist in postgres.